probe-rs-target/src/chip.rs

```diff
@@ -1,6 +1,42 @@
 use super::memory::MemoryRegion;
 use crate::{serialize::hex_option, CoreType};
 use serde::{Deserialize, Serialize};
+
+/// Represents a DAP scan chain element.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct ScanChainDap {
+    /// Unique name of the DAP
+    pub name: Option<String>,
+    /// Specifies the type of the DAP (e.g. ARMCS-DP)
+    pub dap_type: Option<String>,
```

Comment:
I am a bit confused by the type of the DAP. Usually there is 1-N DAPs that have 1-M[n] DPs each. Which then have 1-O[m] APs.
At leats that's what I thought.
It would be great if this was an enum, even if it might not be conclusive at the start.

Reply:
We discussed it a bit in the chat, but the idea I originally had was to distinguish between JTAG chain elements that were debuggable or not. For example in the TI chips that I am working with there is a TAP for the ICEPICK router that is not debuggable, so probe-rs cannot connect to it. I thought to use `device` there. This was also made to match 1:1 with what CMSIS did here:

[scan_chain in CMSIS SDF file](https://open-cmsis-pack.github.io/Open-CMSIS-Pack-Spec/main/html/sdf_pg.html#sdf_element_scanchain). They have DAP and device.

Eventually I thought to keep the fields 1:1 between probe-rs and the SDF file for easy parsing, but I think that makes the code a bit messier and adds unnecessary complexity, probe-rs doesn't really care about any nondebuggable targets outside of their role in the scan chain, so we don't need this enum.

Reply:
Fair enough, that sounds good to me! If we ever need more functionality later on, we can upgrade it. I think it would be interesting to use the ICEPICK router through probe-rs, but it's a bit out of scope for now :)